File: 0401-0500/0485.max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        const int n = nums.size();
        int i = 0, j = 0;
        int best = 0;
        while (i < n) {
            while (i < n && nums[i] != 1) ++i;
            j = i;
            while (j < n && nums[j] == 1) ++j;
            best = max(best, j - i);
            i = j + 1;
        }
        return best;
    }
};

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        const int n = nums.size();
        int cnt = 0;
        int best = 0;
        for (int x : nums) {
            if (x == 0) cnt = 0;
            else {
                best = max(best, ++cnt);
            }
        }
        return best;
    }
};

class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        if (timeSeries.empty()) return 0;
        int ans = 0;
        for (int i = 1; i < timeSeries.size(); ++i) {
            ans += min(duration, timeSeries[i] - timeSeries[i-1]);
        }
        ans += duration;
        return ans;
    }
};

File: 0401-0500/0485-max-consecutive-ones.rs
struct Solution {}

impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let mut ans = 0;
        let mut left = 0;
        while left < nums.len() {
            while left < nums.len() && nums[left] != 1 {
                left += 1;
            }
            // 如果left出界，需要break掉
            if left >= nums.len() {
                break;
            }
            let mut right = left + 1;
            while right < nums.len() && nums[right] == 1 {
                right += 1;
            }
            ans = ans.max(right - left);
            left = right;
        }
        ans as i32
    }
}

fn main() {
    let nums = vec![1, 1, 0, 1, 1, 1];
    assert_eq!(Solution::find_max_consecutive_ones(nums), 3);

    let nums = vec![0];
    assert_eq!(Solution::find_max_consecutive_ones(nums), 0);

    let nums = vec![1];
    assert_eq!(Solution::find_max_consecutive_ones(nums), 1);
}
